### PR TITLE
Tuple parser

### LINKED ISSUE #35: Implement parser for tuples and 3-tuples
Implement parser for tuples and 3-tuples so that the tests created in #34 are green! :green_heart:

### PR DESCRIPTION
**Pull request by halfzebra**
Closes #35

### CODE REVIEW

File: src/compiler/Error.elm [hunk 1]
```diff
@@ -60,6 +60,8 @@ type ParseContext
     | InLambda
     | InList
     | InUnit
+    | InTuple2
```
Comment (Janiczek) [inline src/compiler/Error.elm]: Let's name the normal tuple `Tuple` and the triple `Tuple3` :) No need for the `2`.

Comment (halfzebra) [reply src/compiler/Error.elm]: I've been considering proposing using the notation the original compiler uses and updating all the other types, what do you think?

Comment (Janiczek) [reply src/compiler/Error.elm]: @halfzebra I might be a bit confused now - you mean like this? https://github.com/elm/compiler/blob/master/compiler/src/AST/Source.hs#L66

Comment (halfzebra) [reply src/compiler/Error.elm]: I was thinking about the notation used for [JS constructors](https://github.com/elm/compiler/blob/33f153eea8222b481fcf2f6d42c83b1230d00d06/compiler/src/Generate/JavaScript/Expression.hs#L171).

I'll make it consistent first 🙂

**Comment by Janiczek** [comment]
Thanks, great work! We can probably refactor `backtrackable` out later, but to be fair I tried to implement this without it and failed miserably. So I'm glad for your working version :)

**Comment by halfzebra** [comment]
Thanks for merging! :+1:

**Pull request merged**